[Fix] Minimum Runable code

andyjjrt opened the pull request:
First, thanks for your contribution, this system looks cool.
This PR produce a minimum runnable code, which makes sure docs align with the system and can be setup on a single machine.

Key changes:
- Remove unused variables in environment variable files
- Fix typos of filename and database name
- Fix docs

If there's any error, please let me know!

andyjjrt committed:
fix: Remove unused varialbe, indicate what does variables do.

andyjjrt committed:
fix: Fix critical spelling mistake, change order to init db properly

andyjjrt committed:
fix: Remove unknown memo variable before initialization

andyjjrt committed:
fix: Add missing package

andyjjrt committed:
fix: Fix critical db name error

andyjjrt committed:
fix: Add port support for db and unify DB_HOST

andyjjrt committed:
fix: Remove `#MB` which cause envfile reading error

andyjjrt committed:
fix: Add network host to make frontend reachable inside docker

andyjjrt committed:
fix: Add missing example envfile and add envfile to gitignore

andyjjrt committed:
doc: Update frontend document

andyjjrt committed:
fix: Remove unused config

andyjjrt committed:
fix: Revert createdb.sql and remove it from db-init.js

andyjjrt committed:
doc: Update backend document

andyjjrt committed:
doc: Update backend document

andyjjrt committed:
doc: Update frontend document

andyjjrt committed:
fix: Add back PYTHON_BIN

ccchungdaw reviewed (changes requested):
In the .env.example file, line 47 defines PYTHON_BIN.
For run.sh, it may return a Python binary that is not actually the one being used.

ccchungdaw reviewed (approved):
Thanks a lot:)

ccchungdaw reviewed (approved):
Yes, it would be best to adjust the order to ensure compatibility across different database versions. Thank you!

ccchungdaw reviewed (commented):
I need to make an adjustment here, it should be HF_TOKEN. I'll decline for now.

andyjjrt committed:
fix: Remove duplicated `PYTHON_BIN`

ccchungdaw reviewed (changes requested):
-p ${PORT}:80 \ 

Removing this line will restrict external access to only port 80, which is not recommended.

ccchungdaw reviewed (changes requested):
`DB_HOST` should still use `DB_SERVER`; please check the `.env.example` file.  
Additionally, I will add `DB_PORT` to the `.env.example`.  
Thanks!

ccchungdaw reviewed (approved):
Thanks a lot!

andyjjrt committed:
fix: Rename `DB_HOST` to `DB_SERVER`

ccchungdaw reviewed (approved):
Thanks a lot!

andyjjrt committed:
fix: Rename `DB_HOST` to `DB_SERVER`

ccchungdaw reviewed (approved):
Thanks, I’ll go ahead and make the changes on my side.

ccchungdaw reviewed (approved):
Yeah, adjusting the order sounds like the best move to keep things compatible with different database versions. Thanks!

ccchungdaw reviewed (commented):
just need to clean things up a bit on our side.

andyjjrt committed:
fix: Add back port config and add new command for same host

andyjjrt commented:
> In the .env.example file, line 47 defines PYTHON_BIN. For run.sh, it may return a Python binary that is not actually the one being used.

Duplicated 'PYTHON_BIN' has been removed (a2785a73b9d6072a3b9e64564ea9e36bcdef8589), I didn't notice it. 
And in `run.sh`, `PYTHON_BIN` seems not to be used in this file, that's why I removed it in my version.

> -p ${PORT}:80 \
> 
> Removing this line will restrict external access to only port 80, which is not recommended.

I've added and tested another configuration using docker `--add-host` ([docs](https://docs.docker.com/reference/cli/docker/container/run/#add-host))

> `DB_HOST` should still use `DB_SERVER`; please check the `.env.example` file. Additionally, I will add `DB_PORT` to the `.env.example`. Thanks!

Fixed! (cd7f9175910f807cb18ba1daa1f1161415d6158b, 4511c8189180fa04d42006f954e9d68dd4b991a7)
